json-rpc/src/data.rs

```diff
@@ -0,0 +1,420 @@
+use crate::{
+    errors::JsonRpcError,
+    util::{transaction_data_view_from_transaction, vm_status_view_from_kept_vm_status},
+    views::{
+        AccountStateWithProofView, AccountView, BytesView, CurrencyInfoView, EventView,
+        EventWithProofView, MetadataView, StateProofView, TransactionView, TransactionsProofsView,
+        TransactionsWithProofsView,
+    },
+};
+use anyhow::{format_err, Result};
+use diem_crypto::{hash::CryptoHash, HashValue};
+use diem_types::{
+    account_address::AccountAddress,
+    account_config::{
+        diem_root_address, from_currency_code_string, resources::dual_attestation::Limit,
+        AccountResource,
+    },
+    account_state::AccountState,
+    chain_id::ChainId,
+    event::EventKey,
+    ledger_info::LedgerInfoWithSignatures,
+};
+use network::counters;
+use std::{
+    cmp::min,
+    convert::{TryFrom, TryInto},
+    sync::Arc,
+};
+use storage_interface::{DbReader, Order};
+
+pub fn get_account_state(
+    db: &Arc<dyn DbReader>,
```

Comment:
instead of passing db around, maybe better to have Reader{db: Arc<dyn DbReader>}, and attach all funcs to this struct.

Reply:
yeah I went back and forth on that one- wasn't sure whether to keep them purely functional or not; it would mean modifying the jsonRPC service to pass around the new object along with the DB, which felt like more changes than leaving it functional, but if you want I can make that change, I'm not too opinionated on it either way

Reply:
That's sort of the direction I went with the handler. Is there a reason you can't just construct the handler and call the methods on the handler itself? One other disadvantage of this approach is that I noticed some of the bounds checking on versions didn't translate over to these functions here

Reply:
I think the question is, if someone created a function calls to the service.db, do you want to move that function this this file? If so, then wrap DbReader inside a struct as private field will ensure it happens, otherwise, you will have other db access calls here and there.

Reply:
that’s fair- I think I’d like to leave the functions free for now and potentially refactor later if/when we try to unify jsonrpc/streamrpc